cognite/neat/rules/_shared.py

```diff
@@ -1,19 +1,61 @@
-from typing import TypeAlias, TypeVar
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Generic, TypeAlias, TypeVar
 
+from cognite.neat.issues import IssueList
 from cognite.neat.rules.models import (
     AssetRules,
     DMSRules,
     DomainRules,
     InformationRules,
 )
-from cognite.neat.rules.models.asset._rules_input import AssetRulesInput
-from cognite.neat.rules.models.dms._rules_input import DMSRulesInput
-from cognite.neat.rules.models.information._rules_input import InformationRulesInput
+from cognite.neat.rules.models.asset._rules_input import AssetInputRules
+from cognite.neat.rules.models.dms._rules_input import DMSInputRules
+from cognite.neat.rules.models.information._rules_input import InformationInputRules
 
 VerifiedRules: TypeAlias = DomainRules | InformationRules | DMSRules | AssetRules
-InputRules: TypeAlias = AssetRulesInput | DMSRulesInput | InformationRulesInput
+InputRules: TypeAlias = AssetInputRules | DMSInputRules | InformationInputRules
 Rules: TypeAlias = (
-    AssetRulesInput | DMSRulesInput | InformationRulesInput | DomainRules | InformationRules | DMSRules | AssetRules
+    AssetInputRules | DMSInputRules | InformationInputRules | DomainRules | InformationRules | DMSRules | AssetRules
 )
 T_Rules = TypeVar("T_Rules", bound=Rules)
 T_VerifiedRules = TypeVar("T_VerifiedRules", bound=VerifiedRules)
+T_InputRules = TypeVar("T_InputRules", bound=InputRules)
+
+
+@dataclass
+class OutRules(Generic[T_Rules], ABC):
+    """This is a base class for all rule states."""
+
+    @abstractmethod
+    def get_rules(self) -> T_Rules | None:
```

Comment:
why is this method needed?

Reply:
To have a way to access rules from a generic `OutRules` object. I did not move `rules` up as it has different type hints in `JustRules` and `MaybeRules`.